src/components/TodoItem.tsx

```diff
@@ -0,0 +1,178 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo,
+  deleteTodo: (id: number) => Promise<unknown>,
+  updatedTodo: (newTodo: Todo) => Promise<unknown>,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  deleteTodo,
+  updatedTodo,
+}) => {
+  const [editValue, setEditValue] = useState(todo.title);
+  const [editTodosId, setEditTodosId] = useState(0);
```

Comment:
```suggestion
  const [editingTodoId, setEditTodosId] = useState(0);
```
and fix `set` accordingly
it's 1 todo which is in `editing` state at the moment